include/libtorrent/http_parser.hpp

```diff
@@ -67,6 +67,14 @@ namespace libtorrent {
 			if (i == m_header.end()) return empty;
 			return i->second;
 		}
+		unsigned header_uint(char const* key, unsigned const def_value) const
```

Comment:
My gut reaction is that I don't think parsing an integer from a string should be a member function. This should be a free function. I suppose you also provide the default in case the header isn't found, which in a way is more robust than to parse an empty string. Maybe it could just have a different name, like ``header_int()``, and return a ``std::int64_t``. What should it do on overflow? either throw an exception or return the default.

More fundamentally though, this should not return an unsigned integer. Unsigned integers have modulo 2 arithmetic, and that's not appropriate for any value parsed out of an http header (at least I find it unlikely). The default should also not be ``unsigned``.

It should also not take a c-string. It should take a ``string_view``.